### PR TITLE
PartitionReceiveHandler for connectivity test

### PR DESCRIPTION
**Pull request by philipktlin**
Create a new PartitionReceiveHandler for connectivity test; which will receive the event from event hub and save event as test result in persisted test result storage.

### CODE REVIEW

File: test/modules/TestResultCoordinator/PartitionReceiverHandler.cs [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+                        else
+                        {
+                            Logger.LogDebug($"Message for module [{moduleIdFromEvent}] and device [{this.deviceId}] doesn't contain batch id and/or sequence number.");
+                        }
+                    }
+                }
+            }
+        }
+
+        public Task ProcessErrorAsync(Exception error)
```
Comment (ancaantochi) [inline test/modules/TestResultCoordinator/PartitionReceiverHandler.cs]: >ProcessErrorAsync [](start = 20, length = 17)

Is it used?

File: test/modules/TestResultCoordinator/PartitionReceiverHandler.cs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+        public int MaxBatchSize { get; set; }
+
+        public async Task ProcessEventsAsync(IEnumerable<EventData> events)
+        {
+            if (events != null)
+            {
+                foreach (EventData eventData in events)
+                {
+                    eventData.SystemProperties.TryGetValue(TrackingIdPropertyName, out object trackingIdFromEvent);
```
Comment (varunpuranik) [inline test/modules/TestResultCoordinator/PartitionReceiverHandler.cs]: this cannot be a string?

File: test/modules/TestResultCoordinator/PartitionReceiverHandler.cs (hunk 1 above)
Comment (philipktlin) [reply test/modules/TestResultCoordinator/PartitionReceiverHandler.cs]: it is part of IPartitionReceiveHandler.

File: test/modules/TestResultCoordinator/PartitionReceiverHandler.cs (hunk 2 above)
Comment (philipktlin) [reply test/modules/TestResultCoordinator/PartitionReceiverHandler.cs]: it doesn't do type conversion; need to get object first and cast to string.

**Pull request merged**